corehq/apps/README.rst

```diff
@@ -0,0 +1,191 @@
+   parts of this code, which interfaces with `Formplayer <https://github.com/dimagi/formplayer/>`_
+   to ultimately run functionality in the `commcare-core <https://github.com/dimagi/commcare-core/>`_
+   repo, which is shared with CommCare Mobile.
+domain
+   Domains, called "project spaces" in user-facing content, are the core sandboxes of CommCare. Almost
+   all features and UIs in CommCare are in the context of a domain.
+export
+   Configurations for exporting project data, typically forms or cases, to an excel download.
+hqmedia
+   Multimedia handling, primarily used in applications.
+hqwebapp
+   Core UI code for the HQ site. Includes things like the standard error pages,
+   javascript widgets, login views, etc.
+locations
+   Locations provide a hierarchical way to organize CommCare users and data.
+reports
+   Standard, pre-canned reports to view project data: Submit History, Worker Activity Report, etc.
+reports_core
+   More reporting code.
+sms
+   Features to send SMS and emails via CommCare HQ. Much of the underlying code is in ``corehq.messaging``.
+userreports
+   User-defined reports. Intertwined with other report-related apps.
+users
+   Users of CommCare HQ and/or CommCare mobile. The primary class dealing with users is ``CouchUser``,
+   a representation of the user in couch. ``CouchUser`` has the subclasses ``WebUser`` and ``CommCareUser``
+   to distinguish between admin-type users who primarily use CommCare HQ and data entry users who primarily use
+   CommCare Mobile. The distinction between web and mobile users is blurry, especially with the advent of
+   Web Apps for data entry in HQ.
+
+Secondary Apps
+^^^^^^^^^^^^^^^^^^^^
+These apps are maintained and updated regularly, but are a bit less core than the set above.
+
+case_importer
+   Bulk import of cases.
+custom_data_fields
+   This allows users to add arbitrary data to mobile users, locations, and products, which can then
+   be referenced in applications.
+data_dictionary
+   The data dictionary documents a project's data model, specifically, its case types and case properties.
+   This makes it possible to reference those definitions throughout the app-building process.
+   The data dictionary is used partially for project documentation but is also referenced in a few other
+   parts of HQ: for example, when configuring case properties to be updated in a form, app manager will
+   pull the properties' descriptions from the data dictionary.
+data_interfaces
+   Functionality for taking a few specific actions on project data, such as reassigning cases in bulk.
+   Most of this app relies heavily on standard reporting functionality.
+fixtures
+   "Fixtures" is used generally in the codebase to refer to datasets sent to the phone, which includes the
+   casedb, mobile report data, etc., but the fixtures app primarily deals with one specific type of fixture,
+   which is lookup tables.
```

Comment:
Possible rewrite:

The term "fixtures" comes from the [XML data model](https://github.com/dimagi/commcare-core/wiki/fixtures) used to send custom structured data to the mobile devices (separate from case data). During the sync request with the mobile device various different fixtures may be sent to the device including lookup tables, locations and mobile reports.
This Django app only deals with the "lookup table" fixtures. It provides the UI for creating and editing them and the code to serialize them to XML.

Reply:
fb3b1a17cb8f36a179c2cc87b946c3c0f6f2333a